### PR TITLE
Dependency simplify

### PR DESCRIPTION
**Pull request by sethrj**
- Simplified the code (fewer stub files) by deleting `CELER_CUDA_CALL` content when CUDA is disabled
- Added a similar function for MPI calls (which as a bonus analogy to CUDA now gives error strings on failure)
- Fix the [apparently failing valgrind tests](https://github.com/celeritas-project/celeritas/runs/1906208618) on master

Prerequisite for #149

### CODE REVIEW

File: src/base/Assert.hh [hunk 1]
```diff
@@ -171,6 +185,42 @@
...
+ * When MPI support is enabled, execute the wrapped statement and throw a
+ * RuntimeError if it fails. If MPI is disabled, throw an unconfigured
+ * assertion.
+ *
+ * If it fails, we call \c cudaGetLastError to clear the error code.
+ *
+ * \code
+ *     CELER_CUDA_CALL(cudaMalloc(&ptr_gpu, 100 * sizeof(float)));
+ *     CELER_CUDA_CALL(cudaDeviceSynchronize());
+ * \endcode
```
Comment (amandalund) [inline src/base/Assert.hh:200]: Left in by accident?

**Review by amandalund — APPROVED** [review]
Thanks @sethrj!

File: src/base/Assert.hh (hunk 1 above)
Comment (sethrj) [reply src/base/Assert.hh:200]: Yep, copy-paste error on my part. Clearly I don't read code examples...

File: src/base/Assert.hh [hunk 2]
```diff
@@ -162,6 +167,15 @@
         }                                                \
     } while (0)
 
+#if !CELERITAS_USE_CUDA
```
Comment (pcanal) [inline src/base/Assert.hh:170]: Why
```
#define X
#if !Y
#undef X
#define X
#endif
```
rather than
```
#if Y
#define X
#else
#define X
#endif
```

Comment (sethrj) [reply src/base/Assert.hh:170]: Yeah, now that you mention it that would be more close to how the other macros are defined 😔 . I guess I did it to leave the macro closer to the documentation?

Comment (pcanal) [reply src/base/Assert.hh:170]: > I guess I did it to leave the macro closer to the documentation?

By one line :~) :~)

Another "advantage" of the current version is that the effective version is intended one less time.

**Pull request merged**